build-system/tasks/check-links.js

```diff
@@ -16,88 +16,95 @@
 'use strict';
 
 const argv = require('minimist')(process.argv.slice(2));
-const BBPromise = require('bluebird');
 const fs = require('fs-extra');
 const log = require('fancy-log');
-const markdownLinkCheck = BBPromise.promisify(require('markdown-link-check'));
+const markdownLinkCheck = require('markdown-link-check');
 const path = require('path');
-const {
-  gitDiffAddedNameOnlyMaster,
-  gitDiffNameOnlyMaster,
-} = require('../common/git');
-const {green, magenta, red, yellow} = require('ansi-colors');
+const {getFilesToCheck} = require('../common/utils');
+const {gitDiffAddedNameOnlyMaster} = require('../common/git');
+const {green, cyan, red, yellow} = require('ansi-colors');
 const {isTravisBuild} = require('../common/travis');
+const {linkCheckGlobs} = require('../test-configs/config');
 const {maybeUpdatePackages} = require('./update-packages');
 
+let filesIntroducedByPr;
+
 /**
- * Parses the list of files in argv, or extracts it from the commit log.
- *
- * @return {!Array<string>}
+ * Checks for dead links in .md files passed in via --files or --local_changes.
  */
-function getMarkdownFiles() {
-  if (!!argv.files) {
-    return argv.files.split(',');
+async function checkLinks() {
+  maybeUpdatePackages();
+  if (!isValidUsage()) {
+    return;
   }
-  return gitDiffNameOnlyMaster().filter(function(file) {
-    return path.extname(file) == '.md' && !file.startsWith('examples/');
-  });
+  const filesToCheck = getFilesToCheck(linkCheckGlobs);
+  if (filesToCheck.length == 0) {
+    return;
+  }
+  if (!isTravisBuild()) {
+    log(green('Starting checks...'));
+  }
+  filesIntroducedByPr = gitDiffAddedNameOnlyMaster();
+  const filesWithDeadLinks = [];
```

Comment:
I'm assuming this is not a result of a map so this happens concurrently now?

Reply:
There's still a `.map` on line 50, but this is much faster. The earlier code was calling `gitDiffAddedNameOnlyMaster()` (which runs a `git` command) once per link.